File: hiho/1270/src/P1270.java
import java.util.Scanner;

/**
 * Created by qwordy on 3/24/16.
 * 1270
 */

public class P1270 {
  public static void main(String[] args) {
    int q, i;
    scanner = new Scanner(System.in);
    q = scanner.nextInt();
    for (i = 0; i < q; i++)
      solve();
  }

  static void solve() {
    int n, m, k, t, i, j, cost, min, level, sum;
    n = scanner.nextInt();
    m = scanner.nextInt();
    k = scanner.nextInt();
    t = scanner.nextInt();
    int[] a = new int[m];
    int[] b = new int[m];
    int[] d = new int[k + 1];
    for (i = 0; i < m; i++)
      a[i] = scanner.nextInt();
    for (i = 0; i < m; i++)
      b[i] = scanner.nextInt();
    sum = 0;
    for (level = 0; level < n; level++) {
      for (i = 1; i <= k; i++) {
        min = Integer.MAX_VALUE;
        for (j = 0; j < m; j++) {
          cost = b[j] == 0 ? Integer.MAX_VALUE : (i - b[j] < 0 ? 0 : d[i - b[j]]) + a[j];
          if (cost < min) min = cost;
        }
        d[i] = min;
      }
      if (d[k] == Integer.MAX_VALUE) {
        System.out.println("No Answer");
        return;
      }
      sum += d[k];
      for (i = 0; i < m; i++) b[i] /= t;
    }
    System.out.println(sum);
  }

  static Scanner scanner;
}

File: hiho/1270/P1270.cpp
#include <stdio.h>
const int MAX = ~(1 << 31);

void solve() {
  int n, m, k, t, i, j, cost, min, level, sum;
  int a[100], b[100], d[10001];
  scanf("%d%d%d%d", &n, &m, &k, &t);
  for (i = 0; i < m; i++)
    scanf("%d", a + i);
  for (i = 0; i < m; i++)
    scanf("%d", b + i);
  d[0] = sum = 0;
  for (level = 0; level < n; level++) {
    for (i = 1; i <= k; i++) {
      min = MAX;
      for (j = 0; j < m; j++) {
        cost = b[j] == 0 ? MAX : (i - b[j] < 0 ? 0 : d[i - b[j]]) + a[j];
        //printf("%d\n", cost);
        if (cost < min) min = cost;
      }
      d[i] = min;
    }
    //printf("%d\n", d[k]);
    if (d[k] == MAX) {
      printf("No Answer\n");
      return;
    }
    sum += d[k];
    for (i = 0; i < m; i++) b[i] /= t;
  }
  printf("%d\n", sum);
}

int main() {
  int q, i;
  scanf("%d", &q);
  for (i = 0; i < q; i++)
    solve();
  return 0;
}
